### DISCUSSION TITLE
new connector: Turso Database

### DISCUSSION THREAD

**Discussion by glommer** [airbytehq/airbyte#36183] [New Connector Request]
Hello,

I'd love to have a destination and source (but mostly destination) connector for [Turso](https://turso.tech). I am the CEO of Turso and we're willing to help if needed. In fact, I tried to write a connector myself and I am struggling a bit - so in parallel of asking for help I am following the instructions [here](https://docs.airbyte.com/integrations/custom-connectors/) and just asking =)

If someone gets to this, Turso is basically SQLite in the cloud, so the SQLite connector should work, except a few methods need to be changed to use our drivers, that talk over http.

In fact, here is how I implemented `check`:

```
import libsql_client
[...]

def get_client(config: Mapping[str, Any]) -> libsql_client.Client:
    url = config.get("url")
    token = config.get("token")
    return libsql_client.create_client_sync(url, auth_token=token)
```

```
    def check(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> AirbyteConnectionStatus:
        try:
            client = get_client(config=config)
            query = client.execute("select 1")
            res = query.rows.pop()
            if res.astuple() != (1,):
                raise Exception("response from the database is wrong")

            return AirbyteConnectionStatus(status=Status.SUCCEEDED)
        except Exception as e:
            return AirbyteConnectionStatus(status=Status.FAILED, message=f"An exception occurred: {repr(e)}")
```

My main problem now is with `write`.

**Comment by slavakurilyak** [comment]
+1 for libSQL or Turso integration!

Lack of this destination prevents me from using Airbyte

[a comment by an automated account is not shown]
